src/Symfony/Component/Serializer/Annotation/DiscriminatorMap.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Annotation;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+/**
+ * Annotation class for @DiscriminatorMap().
+ *
+ * @Annotation
+ * @Target({"CLASS"})
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class DiscriminatorMap
+{
+    /**
+     * @var string
+     */
+    private $typeProperty;
+
+    /**
+     * @var array
+     */
+    private $mapping;
+
+    /**
+     * @param array $data
+     *
+     * @throws InvalidArgumentException
+     */
+    public function __construct(array $data)
+    {
+        if (!isset($data['typeProperty']) || !$data['typeProperty']) {
```

Comment:
`array_key_exist` should be used instead of `isset` here

Reply:
Yes, `null` or a falsy contents should throw an exception, hence `isset`.